### PR TITLE
map_of/3 accepts options instead of max_tries

### PR DESCRIPTION
**Pull request by davydog187**

### CODE REVIEW

File: lib/stream_data.ex [hunk 1]
```diff
@@ -1164,12 +1164,14 @@ defmodule StreamData do
...
-  @spec map_of(t(key), t(value), non_neg_integer()) :: t(%{optional(key) => value})
+  @spec map_of(t(key), t(value), keyword()) :: t(%{optional(key) => value})
         when key: term(),
              value: term()
-  def map_of(key_data, value_data, max_tries \\ 10) do
+  def map_of(key_data, value_data, options \\ []) do
+    options = Keyword.put(options, :uniq_fun, fn {key, _value} -> key end)
+
     {key_data, value_data}
-    |> uniq_list_of(uniq_fun: fn {key, _value} -> key end, max_tries: max_tries)
```
Comment (davydog187) [inline lib/stream_data.ex]: `uniq_list_of` already defaults `max_tries` to 10

File: lib/stream_data.ex [hunk 2]
```diff
@@ -1164,12 +1164,14 @@ defmodule StreamData do
   Shrinks towards smallest maps and towards shrinking keys and values according
```
Comment (davydog187) [inline lib/stream_data.ex]: I need to update the docs here to reflect the api change

File: lib/stream_data.ex [hunk 3]
```diff
@@ -1156,20 +1156,22 @@ defmodule StreamData do
 
   ## Examples
 
-      Enum.take(StreamData.map_of(StreamData.integer(), StreamData.boolean()), 3)
+      Enum.take(StreamData.map_of(StreamData.integer(), StreamData.boolean()), max_tries: 3)
```
Comment (michalmuskala) [inline lib/stream_data.ex]: The `3` argument was to `Enum.take/2`, so it was correct

Comment (davydog187) [reply lib/stream_data.ex]: Good catch.

File: lib/stream_data.ex (hunk 2 above)
Comment (whatyouhide) [reply lib/stream_data.ex]: @davydog187 no worries, I quickly pushed an updated to the docs (and fixed the other comment from Michal). :)

**Comment by whatyouhide** [comment]
Thanks a lot @davydog187, great job! 💟

**Pull request merged**